### DISCUSSION TITLE
pyright and signaturehelp/documentation

### DISCUSSION THREAD

**Discussion by jdtsmith** [joaotavora/eglot#904] [Q&A]
I recently switched back to eglot and am enjoying the simplicity again; thanks to all the devs for the continuous improvements.  In the meantime, I switched my python LSP server to pyright from pylsp.  Sadly I've lost a few things:

1. Signature help shown the echo area when in a function signature does not include the function name (e.g. `(data: Unknown, dtype: Unknown = ...) -> Unknown`)
2.  Arguments are no longer highlighted by position within the argument list (ala elisp-mode).  Formerly if in the first position, `data` above would be highlighted.
3. `eldoc-doc-buffer` does not display the documentation just the (skimpy) signature.

I know the server is returning information on the parameter positions and full documentation.  E.g. here for `np.asmatrix`:

``` 
(:jsonrpc "2.0" :id 17 :result
	  (:signatures
	   [(:label "(data: Unknown, dtype: Unknown = ...) -> Unknown" :parameters
		    [(:label
		      [1 14])
		     (:label
		      [16 36])]
		    :documentation
		    (:kind "plaintext" :value "Interpret the input as a matrix.\n\nUnlike `matrix`, `asmatrix` does not make a copy if the input is already\na matrix or an ndarray.  Equivalent to ``matrix(data, copy=False)``.\n\nParameters\n----------\ndata : array_like\n    Input data.\ndtype : data-type\n   Data-type of the output matrix.\n\nReturns\n-------\nmat : matrix\n    `data` interpreted as a matrix.\n\nExamples\n--------\n>>> x = np.array([[1, 2], [3, 4]])\n\n>>> m = np.asmatrix(x)\n\n>>> x[0,0] = 5\n\n>>> m\nmatrix([[5, 2],\n        [3, 4]])")
		    :activeParameter 0)]
	   :activeSignature 0 :activeParameter 2))
```

**Comment by joaotavora** [comment]
This would seem to be a bug in `eglot--sig-info`.  I'd start by tracing or stepping through that function.
